docs-ref-conceptual/get-started-tutorial-3-use-variables.md

```diff
@@ -0,0 +1,305 @@
+
+# Get a list of all Azure storage accounts that allow blob public access.
+# Notice the backticks and escape characters needed for boolean values.
+az storage account list --resource-group $rgName \
+                        --query "[?allowBlobPublicAccess == \`true\`].name"
+
+# Get a list of Azure storage accounts that were created in the last 30 days. Return the results as a table.
+saDate=$(date +%F -d "-30days")
+az storage account list --resource-group $rgName \
+                        --query "[?creationTime >='$saDate'].{saName:name, createdTimeStamp:creationTime}" \
+                        --output table
+
+# Get a list of Azure storage accounts created in this tutorial
+az storage account list --resource-group $rgName \
+                        --query "[?contains(name, 'msdocs')].{saName:name, saKind:kind, saPrimaryLocation:primaryLocation,    createdTimeStamp:creationTime}" \
+                        --output table
+```
+
+# [PowerShell](#tab/powershell)
+
+```azurecli-interactive
+rgName="<msdocs-tutorial-rg-00000000>"
+
+# Get a list of all Azure storage accounts that allow blob public access.
+az storage account list --resource-group $rgName `
+                        --query "[?allowBlobPublicAccess == ``true``].name"
+
+# Get a list of Azure storage accounts that were created in the last 30 days. Return the results as a table.
+$saDate=Get-Date
+$saDate=$saDate.AddDays(-30).tostring("yyyy-mm-dd")
+az storage account list --resource-group $rgName `
+                        --query "[?creationTime >='$saDate'].{saName:name, createdTimeStamp:creationTime}" `
+                        --output table
+
+# Get a list of Azure storage accounts created in this tutorial
+az storage account list --resource-group $rgName `
+                        --query "[?contains(name, 'msdocs')].{saName:name, saKind:kind, saPrimaryLocation:primaryLocation,    createdTimeStamp:creationTime}" `
+                        --output table
+```
+
+---
+
+## Create a new Azure resource storing output in a variable
+
+Learning to store command output in a variable is beneficial when creating Azure resources which output secrets that should be protected. For example, when you create a service principal, reset a credential, or get an Azure key vault secret, the command output should be protected.
+
+Create a new Azure Key Vault and secret returning command output to a variable. Your Azure Key Vault name must be globally unique, so the `$RANDOM` identifier is used in this example. For more Azure Key Vault naming rules, see [Common error codes for Azure Key Vault](/azure/key-vault/general/common-error-codes).
+
+# [Bash](#tab/bash)
+
+```azurecli-interactive
+# Set your variables.
+let "randomIdentifier=$RANDOM*$RANDOM"
+rgName=<msdocs-tutorial-rg-00000000>
+kvName=msdocs-kv-$randomIdentifier
+location=eastus
+
+# Set your default output to none
+az config set core.output=none
+
+# Create a new Azure Key Vault returning the Key Vault ID
```

Comment:
Not sure if it would be helpful to add a small note saying that this process might take a minute or 2.

Reply:
good suggestion.  I added a note.